File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.js
/**
 * @param {number[]} cardPoints
 * @param {number} k
 * @return {number}
 */
var maxScore = function(cardPoints, k) {
        k = cardPoints.length-k;
        let maxSum = 0;
        let i=0;
        let j =k;
        let totalSum = 0;
        for(let i=0;i<k;i++){
            totalSum+=cardPoints[i];
            maxSum +=cardPoints[i];
        }
        let cur = maxSum;
        while(j<cardPoints.length){
            totalSum+=cardPoints[j];
            cur = cur - cardPoints[i] + cardPoints[j];
            maxSum = Math.min(maxSum,cur);
            i++;
            j++;
        }
        return totalSum - maxSum;
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        k = cardPoints.size()-k;
        int maxSum = 0;
        int i=0;
        int j =k;
        int totalSum = 0;
        for(int i=0;i<k;i++){
            totalSum+=cardPoints[i];
            maxSum +=cardPoints[i];
        }
        int cur = maxSum;
        while(j<cardPoints.size()){
            totalSum+=cardPoints[j];
            cur = cur - cardPoints[i] + cardPoints[j];
            maxSum = min(maxSum,cur);
            i++;
            j++;
        }
        return totalSum - maxSum;
    }
};